packages/server-wallet/src/__chain-test__/directly-funded-channel.test.ts

```diff
@@ -139,143 +177,31 @@ it.skip('Create a directly funded channel between two engines ', async () => {
-    status: 'opening',
-    turnNum: 0,
-  });
-
-  const postFundA = await postFundAPromise;
-  const postFundB = await postFundBPromise;
-
-  expect(postFundA.channelResult).toMatchObject({
-    channelId,
-    status: 'opening',
-    turnNum: 0,
-  });
-
-  expect(postFundB.channelResult).toMatchObject({
-    channelId,
-    status: 'opening',
-    turnNum: 0,
-  });
-
-  await b.pushMessage(getPayloadFor(participantB.participantId, postFundA.outbox));
-
-  const resultA1 = await a.pushMessage(getPayloadFor(participantA.participantId, postFundB.outbox));
-  expect(getChannelResultFor(channelId, resultA1.channelResults)).toMatchObject({
-    status: 'running',
-    turnNum: 3,
-  });
-
-  const resultB1 = await b.pushMessage(getPayloadFor(participantB.participantId, postFundA.outbox));
-  expect(getChannelResultFor(channelId, resultB1.channelResults)).toMatchObject({
-    status: 'running',
-    turnNum: 3,
-    // TODO: the fundingStatus is incorrect as the funding table is not joined with channels table
-    //  when processing the pushMessage above
-    // fundingStatus: 'Funded',
-  });
-
-  const closeA = await a.closeChannel({channelId});
-  expect(closeA.channelResult).toMatchObject({
-    status: 'closing',
-    turnNum: 4,
-    // TODO: the fundingStatus is incorrect as the funding table is not joined with channels table
-    //  when processing the pushMessage above
-    // fundingStatus: 'Funded',
-  });
-
-  const closeB = await b.pushMessage(getPayloadFor(participantB.participantId, closeA.outbox));
-  expect(getChannelResultFor(channelId, closeB.channelResults)).toMatchObject({
-    status: 'closed',
-    turnNum: 4,
-    fundingStatus: 'Defunded',
-  });
-
-  const close2A = await a.pushMessage(getPayloadFor(participantA.participantId, closeB.outbox));
-  expect(getChannelResultFor(channelId, close2A.channelResults)).toMatchObject({
-    status: 'closed',
-    turnNum: 4,
-    fundingStatus: 'Funded',
-  });
-
-  // Mine a few blocks, but not enough for the chain service to update holdings
-  // Then wait 500ms so that, if the chain service incorrectly updated holdings,
-  // then the updated holdings would have been processed by the engine.
-  await mineBlocks(3);
-  await new Promise(r => setTimeout(r, 500));
-  expect((await a.getState({channelId})).channelResult).toMatchObject({
-    status: 'closed',
-    turnNum: 4,
-    fundingStatus: 'Funded',
-  });
-
-  await mineBlocks(2);
-
-  const channelClosedA = await channelClosedAPromise;
-
-  expect(channelClosedA.channelResult).toMatchObject({
-    status: 'closed',
-    turnNum: 4,
-    fundingStatus: 'Defunded',
-  });
+  const {channelId} = response[0];
+  const closeResponse = await b.closeChannels([channelId]);
+  await expect(closeResponse).toBeObjectiveDoneType('Success');
 
   const aBalanceFinal = await getBalance(aAddress);
   const bBalanceFinal = await getBalance(bAddress);
 
   expect(BN.sub(aBalanceFinal, aBalanceInit)).toEqual(aFunding);
```

Comment:
It might be worth it to set `aFunding` and `bFunding` to something more interesting that 1 and 0. Maybe 2 and 3? This would force `b` to make a deposit.

Reply:
I did this in f42a5a9 which broke the tests! Thanks for bringing this up, investigating now.

Reply:
Ah it looks like it was due to the fact that we only `mine` blocks on the `Deposited` event. Since the `Deposited` event only fires when depositing funds, there's nothing to trigger mining a bunch of blocks when conclude is called.

   I'm guessing that with only one deposit the mining from the `Deposited` event was close enough to mine the conclude transaction as well so that's why we didn't see this with `aFunding=1, bFunding=0`.

This is fixed in https://github.com/statechannels/statechannels/pull/3557/commits/6f304b73de94b1ee13b763ae5d8095cad0e50218 by mining on the allocationUpdated event as well.

Reply:
If there was a single deposited event, in theory, the test should have failed. If I recall correctly, the test is set up to mine the minimum number of blocks for the test to pass.

Reply:
> If there was a single deposited event, in theory, the test should have failed. If I recall correctly, the test is set up to mine the minimum number of blocks for the test to pass.

Ah you're right!

I think the reason it was passing was because `b` was the one calling `closeChannel`. Since it had `0` funding it would succeed as soon as the conclude proof was signed.

Reply:
I've added some test cases in https://github.com/statechannels/statechannels/pull/3557/commits/40c47f5a0e3374fc5fbda10668010e4b66bc6e47 so we now test both A and B closing. I also threw in some cases with messages being dropped / delayed.